### PR TITLE
Internationalization

### PR DESCRIPTION
**Pull request by wilkus27**
Issue: https://github.com/openstax/cnx-recipes/issues/1011

### CODE REVIEW

**Comment by wilkus27** [comment]
@helenemccarron I started with removing order assumption from mixin related with sections titles. 

First, I wanted to create some config in which we could decide about order of all spans with specific classes which create title:

```
$Config_SectionTitle(os-number: ... , os-divider: ' ', os-text: content());
```
But at this moment class `os-text` is added by mixin:

```
@mixin modify_spanWrapTitles() {
  :not([data-type="metadata"]) > [data-type="document-title"] {
    content: '';
    &::after {
      content: content();
      container: span;
      class: "os-text";
    }
  }
}
```
which has to be included in some pass before that one in mixin `number_numberSections` is included.

In 72d1990 I'm trying to create config in which we can decide of order of two parts: existing one - created in content and new one - created by recipe. 

Mixin which I created (`utils_titlePartsOrder`) is changing output:

image

At this moment it's creating content for each key. Probably it's easy, but I couldn't figure out how can I take contents from each loop, and then combine them to one.

**Comment by wilkus27** [comment]
@helenemccarron in 72da88c there is my second proposition. It's not so complex as first one and it doesn't change any output. I aplied those changes only for section titles. I want to be sure we want to go in this way before I will try to apply it for other titles.

**Comment by wilkus27** [comment]
I start to analyze captions of tables and figures. Their content is created in that way:

image

At this moment only first part is defined by config.
So I need to find solution similar to the first one, which I pushed. Because in tables captions there are a few parts I will create some function. In this case it's not a good idea to use  previous solution:

image

**Comment by wilkus27** [comment]
@helenemccarron  @tomjw64 in 6eea7e8 I created new file with order functions. For this moment only for figure and table captions and for chapter, appendix and section titles. I put it in mixins directory, but maybe I should move it somewhere else. Sorry for the mess in this PR, I wanted to try different approaches. Only last one commit is important. Please give me your thoughts about this idea. I want to know your opinion before I start working on functions for other elements.

File: recipes/mixins/_order.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    $contentValue: null;
+
+    @each $titlePart in map-keys($documentTitleConfig) {
+        $titleValue: map-get($documentTitleConfig, $titlePart);
+        $value: null;
+
+        @if $titleValue == 'partTypeTitleConfig' {
+            $value: pending(b#{$element}Label);
+        }
+        @if $titleValue == 'existingContent' {
```
Comment (tomjw64) [inline recipes/mixins/_order.scss]:
```suggestion
        @else if $titleValue == 'existingContent' {
```

`$value` is not guaranteed to be not null after this block, but null is also not an acceptable value for `$value` given how it is used.

File: recipes/books/_common-stuff.scss [hunk 2]
```diff
@@ -14,13 +14,22 @@ $Config_PartType_Chapter_TitleContent   : (os-number: counter(chapter), os-divid
 $Config_PartType_Appendix_TitleContent  : (os-number: counter(appendix, upper-alpha), os-divider: ' | ');
 $Config_PartType_Section_TitleContent   : (os-number: counter(chapter) "." counter(section), os-divider: ' ' );
 
+$Config_Chapter_TitleContentOrder: (Label: partTypeTitleConfig, Content: existingContent);
+$Config_Appendix_TitleContentOrder: (Label: partTypeTitleConfig, Content: existingContent);
+$Config_Section_TitleContentOrder: (Label: partTypeTitleConfig, Content: existingContent);
```
Comment (tomjw64) [inline recipes/books/_common-stuff.scss]: It looks like the only thing this gives us the ability to do is to put the label before or after the existing title. Is this enough? What if something like a circumfix is necessary around the title?

File: recipes/mixins/_order.scss (hunk 1 above)
Comment (tomjw64) [reply recipes/mixins/_order.scss]: Perhaps also an `@else` block that errors when an acceptable `$titleValue` is not given would be nice as well.

File: recipes/mixins/_order.scss [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            $value: content();
+        }
+
+        $contentValue: $contentValue $value;
+    }
+
+    @return $contentValue;
+}
+
+@function order_figTableCaption($figTableConfig) {
```
Comment (tomjw64) [inline recipes/mixins/_order.scss]: The logic of this and `order_documentTitle` are extremely similar. Is there a chance they could be merged somehow?

**Comment by wilkus27** [comment]
In 4679792 I created one function for titles and captions but with assumption that structure of `$Config`s is the same like previously. 

I was thinking also about this comment: https://github.com/openstax/cnx-recipes/pull/1159#discussion_r301180631.

And at this moment I think it is enough because recipes don't support other elements.
In titles there are only spans with classes: `os-number, os-divider, os-text`. 
In captions there are: `os-title-label, os-number, os-divider, os-title, os-divider, os-caption`.
If someone wants to add other element he can do it in two ways:

- by creating its content by himself

Define it in some : 
`$Config_CustomPart: (classNameofSpan: contentOfSpan, classNameofSpan: contentOfSpan);`

Then include utils_title mixin in mixin related with title, like for section titles:

image

- by moving some existing content

For example, like in setCaptionHelper mixin:

image

In both cases he has to modify in some way mixins and add new names for a buckets where content will be moved. Because in order function I'm using names of buckets, either I have to change function to be independent of buckets (what will be hard) or I have to think of the names of the future backets and support them in function.

**Comment by wilkus27** [comment]
**Decision:** this proposition will not be used as a solution for https://github.com/openstax/cnx-recipes/issues/1011. This PR could be closed.

**Pull request closed without merging**